### ISSUE TITLE
Connection issue

### ISSUE THREAD

**Issue by BinarySoftware** [enso-org/ide#774]
<img width="654" alt="Zrzut ekranu 2020-08-31 o 12 02 05" src="https://user-images.githubusercontent.com/26655166/91718040-eb150f80-eb92-11ea-9c1d-c17af3fcb711.png">

### General Summary
As you can see, somehow ports show that a(1) is b(2) and b is a, and whatsmore, a is connected to ',' .

**Comment by wdanilo** [comment]
I made some investigation here and it looks to me like parser <-> span-tree bug. CC @mwu-tow @farmaazon . Consider the expression `[aa, bbb]`, it gives us the following (clearly wrong) span-tree (simplified for clarity):

```rust
SpanTree {
root: Node {
    kind: Root,
    size: Size {
        value: 9,
    },
    children: [
        Child {
            node: Node {
                kind: Argument {
                    is_removable: false,
                },
                size: Size {
                    value: 3,
                },
                children: [],
                expression_id: Some(
                    c26bf3ff-1e5b-4b00-ab56-d944e692bfd9,
                ),
                parameter_info: None,
            },
            offset: Size {
                value: 2,
            },
        },
        Child {
            node: Node {
                kind: Argument {
                    is_removable: false,
                },
                size: Size {
                    value: 1,
                },
                children: [],
                expression_id: Some(
                    50a85dd2-94ac-426e-b3cf-3acab4ab94e7,
                ),
                parameter_info: None,
            },
            offset: Size {
                value: 5,
            },
        },
        Child {
            node: Node {
                kind: Argument {
                    is_removable: false,
                },
                size: Size {
                    value: 2,
                },
                children: [],
                expression_id: Some(
                    8f30835d-a43b-4068-8f06-5b78127cb36f,
                ),
                parameter_info: None,
            },
            offset: Size {
                value: 6,
            },
        },
    ],
    expression_id: Some(
        04d4ba1f-2359-4f3a-8bdc-ca0c1791b2b1,
    ),
    parameter_info: None,
},
```

I feel somehow that the arguments got traversed backwards, and thus these strange results, but I have no idea why. It seems that the second argument is presented before the first (see the span of those).

I made a branch with this behavior embedded into `interfce.rs`, see the `wip/wd/debug-774` branch. Compile it with `./run watch --crate ide/view`, and run `http://localhost:8080/debug/interface`. The spantree will be printed to terminal and you can play with wrong port representation in the browser. Ping me if I could further help here. Here is the exact debug commit: https://github.com/enso-org/ide/commit/351c15de856edce2785d3a2da1fb682dffb1d345

**Comment by sylwiabr** [comment]
![image](https://user-images.githubusercontent.com/12892578/97968259-b43fbd80-1dbe-11eb-9250-5682631e5b04.png)
on version:
core     : 2.0.0-alpha.0
build    : 6daa8c0
electron : 8.1.1
chrome   : 80.0.3987.141
@farmaazon

**Comment by farmaazon** [comment]
You tested int on the commit before fix was delivered: 17e4190682ba97b94e943e9310711f2d9c7d8622

**Comment by sylwiabr** [comment]
True
